code/helpers/vector.dm

```diff
@@ -93,6 +106,9 @@ Inputs:
 			if((owner.loc == target_turf) || T.check_thrown_collision(owner) || !owner.Move(T))
 				owner.pixel_x = initial_pixel_x
 				owner.pixel_y = initial_pixel_y
+				owner.transform = null
+				owner.update_icon()
```

Comment:
I'm not convinced calling update_icon every move (20 times per second) is a good idea.